addons/account/models/account_move.py

```diff
@@ -1538,10 +1539,10 @@ def _fetch_duplicate_supplier_reference(self, only_posted=False):
 
         move_table_and_alias = "account_move AS move"
         place_holders = {}
-        if not moves.ids:
+        if not moves.ids or isinstance(moves.id, models.NewId):
```

Comment:
Don't you just need flushing before the query?
This part was intended only for when no flushing is possible (because the record doesn't exist)

Reply:
> maybe it's because we use SQL and not the orm

Flushing means making the SQL server aware of all the changed done in the ORM's cache. So once it is flushed there should be no differences between ORM and SQL